src/vivarium_gates_nutrition_optimization_child/components/causes.py

```diff
@@ -1,17 +1,81 @@
 """
 Component to include birth prevalence in SIS model.
 """
-
+import pandas as pd
 from vivarium.framework.state_machine import State
-from vivarium_public_health.disease import DiseaseModel, DiseaseState
+from vivarium_public_health.disease import DiseaseModel as DiseaseModel_
+from vivarium_public_health.disease import DiseaseState
 from vivarium_public_health.disease import (
     RiskAttributableDisease as RiskAttributableDisease_,
 )
 from vivarium_public_health.disease import SusceptibleState
 from vivarium_public_health.disease.transition import TransitionString
 
 
-def SIS_with_birth_prevalence(cause: str) -> DiseaseModel:
+class DiseaseModel(DiseaseModel_):
+    @property
+    def columns_required(self):
+        return ["age", "sex", "alive", "tracked"]
+
+    def setup(self, builder):
+        super().setup(builder)
+        if "variable_step_sizes" in self._get_data_functions:
+            for state, step_size in self._get_data_functions["variable_step_sizes"].items():
+                builder.time.register_step_modifier(
+                    lambda index: self.modify_step(state, step_size, index)
+                )
+
+    def modify_step(self, state, step_size, index):
+        infected = self.population_view.get(
+            index,
+            f"{self.state_column} == '{state}' and alive == 'alive' and tracked == True",
+        ).index
+        return pd.Series(pd.Timedelta(days=step_size), index=infected)
+
+
+def SIS(cause: str, step_size_days: str = None) -> DiseaseModel:
+    healthy = SusceptibleState(cause, allow_self_transition=True)
+    infected = DiseaseState(cause, allow_self_transition=True)
+
+    healthy.add_rate_transition(infected)
+    infected.add_rate_transition(healthy)
+
+    infected_step_size_function = (
+        {"variable_step_sizes": {infected.state_id: float(step_size_days)}}
+        if step_size_days
+        else {}
+    )
+
+    return DiseaseModel(
+        cause, states=[healthy, infected], get_data_functions=infected_step_size_function
+    )
+
+
+def SIS_fixed_duration(cause: str, duration: str, step_size_days: str = None) -> DiseaseModel:
+    duration = pd.Timedelta(days=float(duration) // 1, hours=(float(duration) % 1) * 24.0)
+
+    healthy = SusceptibleState(cause, allow_self_transition=True)
+    infected = DiseaseState(
+        cause,
+        get_data_functions={"dwell_time": lambda _, __: duration},
+        allow_self_transition=True,
+    )
+
+    healthy.add_rate_transition(infected)
+    infected.add_dwell_time_transition(healthy)
+
+    infected_step_size_function = (
+        {"variable_step_sizes": {infected.state_id: float(step_size_days)}}
```

Comment:
this doesn't match the typing in DiseaseModel (https://github.com/ihmeuw/vivarium_public_health/blob/main/src/vivarium_public_health/disease/model.py#L68).

Shouldn't the values be a callable?

Reply:
Good catch. This should probably be done, though it's not critical, since this implementation is likely temporary.

Reply:
i fixed this in a lazy way (lambda'ing the dictionary) in the interest of coming back to the details here after the new implementation of variable clocks is in place